ros2topic/ros2topic/verb/pub.py

```diff
@@ -115,6 +120,11 @@ def publisher(
     if not isinstance(values_dictionary, dict):
         return 'The passed value needs to be a dictionary in YAML format'
 
+    if once:
+        if times > 0:
+            return 'Cannot pass both -1 and -t <times>'
```

Comment:
This check should happen and `parse_args` as part of a custom type function.

Reply:
I ended up moving this check into main, alongside the other check for args.rate <= 0.

There is actually further work that could be done here to check for errors earlier.  For instance, we could look up the message type/module and fill out the message all within main, before passing it down to the `publish` function.  That would do less setup/teardown work in the case of an error.  But I'll leave that for a future cleanup.

Reply:
I would still suggest to move both checks into custom type functions to determine valid input at argument parsing time.

Reply:
As far as I can tell, this isn't possible.  You can define a custom [type callable](https://docs.python.org/3/library/argparse.html#type) in argparse, but that callable does not have access to the overall `args` structure.  Even if it did, there is no way of knowing that you parsed all of the arguments by the time you got there.  You can really only do this particular check at the end.

That being said, I did implement a custom type check for both rate and times that ensures that that they are positive float and non-negative integer, respectively.

Reply:
I was referring to both individual checks which have been moved into type functions :+1:

If both arguments can't be passed at the same time the arguments should be added to an exclusive group.

Reply:
Good call about the function names; fixed now.  Also, I didn't know about argparse's mutually exclusive groups, thanks for that.  I've now changed it to use them.  I'll run another CI here, and this is ready for another review.